src/Appwrite/Platform/Tasks/GetMigrationStats.php

```diff
@@ -0,0 +1,186 @@
+                    $dbForProject->setNamespace('_' . $project->getInternalId());
+
+                    /** Get Project ID */
+                    $stats['Project ID'] = $project->getId();
+
+                    /** Get Migration details */
+                    $migrations = $dbForProject->find('migrations', [
+                        Query::limit(500)
+                    ]);
+
+                    $migrations = array_map(function ($migration) use ($project) {
+                        $result['Project ID'] = $project->getId();
+                        $result['$id'] = $migration->getAttribute('$id');
+                        $result['$createdAt'] = $migration->getAttribute('$createdAt');
+                        $result['status'] = $migration->getAttribute('status');
+                        $result['stage'] = $migration->getAttribute('stage');
+                        $result['source'] = $migration->getAttribute('source');
+                        return array_values($result);
+                    }, $migrations);
+
+                    if(!empty($migrations)) {
+                        $csv->insertAll($migrations);
+                    }
+                } catch (\Throwable $th) {
+                    Console::error('Failed on project ("' . $project->getId() . '") with error on File: ' . $th->getFile() . '  line no: ' . $th->getline() . ' with message: ' . $th->getMessage());
+                } finally {
+                    $pools
+                        ->get($db)
+                        ->reclaim();
+                }
+            }
+
+            $sum = \count($projects);
+
+            $projects = $dbForConsole->find('projects', [
+                Query::limit($limit),
+                Query::offset($offset),
+            ]);
+
+            $offset = $offset + $limit;
+            $count = $count + $sum;
+        }
+
+        Console::log('Iterated through ' . $count - 1 . '/' . $totalProjects . ' projects...');
+
+        $pools
+            ->get('console')
+            ->reclaim();
+
+        /** @var PHPMailer $mail */
+        $mail = $register->get('smtp');
+
+        $mail->clearAddresses();
+        $mail->clearAllRecipients();
+        $mail->clearReplyTos();
+        $mail->clearAttachments();
+        $mail->clearBCCs();
+        $mail->clearCCs();
+
+        try {
+            /** Addresses */
+            $mail->setFrom(App::getEnv('_APP_SYSTEM_EMAIL_ADDRESS', APP_EMAIL_TEAM), 'Appwrite Cloud Hamster');
+            $recipients = explode(',', App::getEnv('_APP_USERS_STATS_RECIPIENTS', ''));
+
+            foreach ($recipients as $recipient) {
+                $mail->addAddress($recipient);
+            }
+
+            /** Attachments */
+            $mail->addAttachment($this->path);
+
+            /** Content */
+            $mail->Subject = "Migration Report for {$this->date}";
+            $mail->Body = "Please find the migration report atttached";
+            $mail->send();
```

Comment:
Do these tasks have couroutine hook enabled? If so, has the [fix we added for workers](https://github.com/appwrite/appwrite/pull/6966/files) been added?

Reply:
This is a task. so its not required here 👍